File: DnD.Shared/Models/CreateCharacterRequestModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DnD.Shared.Models
{
    public class CreateCharacterRequestModel
    {
        public string Name { get; set; }
        public string Type { get; set; }
        public string ClassId { get; set; }
        public string RaceId { get; set; }
        public bool Visible { get; set; }

    }
}

File: DnD.JavaScript.SDK/src/models/CreateCharacterRequestModel.ts
import ArsenalModel from "./ArsenalModel";
import GearModel from "./GearModel";
import PropertyModel from "./PropertyModel";
import SkillModel from "./SkillModel";
import StatModel from "./StatModel";

export default interface CreateCharacterRequestModel {
    name: string;
    type: string;
    classId: string;
    raceId: string;
    arsenal: ArsenalModel[] | null;
    gear: GearModel[] | null;
    skills: SkillModel[] | null;
    feats: string[] | null;
    specialAbilities: string[] | null;
    stats: StatModel[] | null;
    properties: PropertyModel[] | null;
    visible: boolean;
}